src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averageInArray(person: Student): number {
+  return person.grades.reduce((accum, curr) => accum + curr, 0)
+  / person.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name: {
+        return (order === 'asc')
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name);
+      }
+
+      case SortType.Surname: {
+        return (order === 'asc')
+          ? a.surname.localeCompare(b.surname)
+          : b.surname.localeCompare(b.surname);
+      }
```

Comment:
You duplicate logic here. Check 3 point in the [checklist](https://github.com/mate-academy/ts_sort_students/blob/master/checklist.md)
You have the enum `SortType` and function parameter `sortBy` try to implement it without repeating code.

Hint: You need to change the enum 
```suggestion
export enum sorType {
  Name = 'name',
  Surname = 'surname',
  Age = 'age',
  Married = 'married',
  AverageGrade = 'grades',
}
```